spec/lib/puppet_spec/files.rb

```diff
@@ -28,7 +28,7 @@ def tmpfile(name, dir = nil) PuppetSpec::Files.tmpfile(name, dir) end
   def self.tmpfile(name, dir = nil)
     # Generate a temporary file, just for the name...
     source = dir ? Tempfile.new(name, dir) : Tempfile.new(name)
-    path = source.path.encode(Encoding::UTF_8)
+    path = Puppet::FileSystem.expanded_path_string(source.path.encode(Encoding::UTF_8))
```

Comment:
Need to make sure this doesn't significantly impact test runtimes.  On Windows in a VM, no perceptible difference - but need to look at AppVeyor, Travis, local OSX, etc.